onyo/commands/tests/test_new.py

```diff
@@ -86,6 +86,98 @@ def test_new_top_level(repo: OnyoRepo) -> None:
     fsck(repo)
 
 
+@pytest.mark.repo_dirs(*directories)
+def test_new_from_root_with_absolute_paths(repo: OnyoRepo) -> None:
+    """
+    Test that `onyo new --path <path>` can create an asset with absolute paths
+    while in the root of the repository.
+    """
+    file = 'laptop_apple_macbookpro.faux'
+
+    for d in directories:
+        # build an absolute path for a new asset
+        new_file = Path(f"{d}/{file}").resolve()
+
+        ret = subprocess.run(['onyo', 'new', '--yes', '--path', new_file],
+                             capture_output=True, text=True)
+
+        # verify correct output
+        assert "The following will be created:" in ret.stdout
+        assert str(d) in ret.stdout
+        assert str(new_file) in ret.stdout
+        assert not ret.stderr
+        assert ret.returncode == 0
+
+    # verify that all new assets exists and the repository is in a clean state
+    assert len(repo.asset_paths) == len(directories)
+    fsck(repo)
+
+
+@pytest.mark.repo_dirs(*directories)
+def test_new_sub_dir_absolute_path(repo: OnyoRepo) -> None:
+    """
+    Test that `onyo new --path <path>` can create an asset in sub-directories of
+    the repository while the cwd is inside a different sub-directory.
+    """
+    file = 'laptop_apple_macbookpro.faux'
+
+    for cwd in directories:
+        # change into a sub-directory of the repository
+        cwd = repo.git.root / cwd
+        os.chdir(cwd)
+        for d in directories:
+            # build an absolute path for a new asset
+            new_file = f"{repo.git.root}/{d}/{file}"
+
+            ret = subprocess.run(['onyo', 'new', '--yes', '--path', new_file],
+                                 capture_output=True, text=True)
+
+            # verify correct output
+            assert "The following will be created:" in ret.stdout
+            assert str(d) in ret.stdout
+            assert new_file in ret.stdout
+            assert not ret.stderr
+            assert ret.returncode == 0
+
+    # verify that all new assets exists and the repository is in a clean state
+    assert len(repo.asset_paths) == len(directories) * len(directories)
+    fsck(repo)
+
+
+@pytest.mark.repo_dirs(*directories)
+def test_new_sub_dir_relative_path(repo: OnyoRepo) -> None:
+    """
+    Test that `onyo new --path <path>` can create an asset in sub-directories of
+    the repository while the cwd is inside a different sub-directory with a
+    relative path.
+    """
+    file = 'laptop_apple_macbookpro.faux'
+
+    for cwd in directories:
```

Comment:
Same as above: Nested loop for all combinations of directories seems unnecessarily expensive.